plugin/src/main/cpp/classes/openxr_hybrid_app.cpp

```diff
@@ -0,0 +1,114 @@
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "classes/openxr_hybrid_app.h"
+
+#include <godot_cpp/classes/engine.hpp>
+#include <godot_cpp/classes/os.hpp>
+#include <godot_cpp/core/class_db.hpp>
+
+OpenXRHybridApp *OpenXRHybridApp::singleton = nullptr;
+
+void OpenXRHybridApp::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("is_hybrid_app"), &OpenXRHybridApp::is_hybrid_app);
+	ClassDB::bind_method(D_METHOD("get_mode"), &OpenXRHybridApp::get_mode);
+	ClassDB::bind_method(D_METHOD("switch_mode", "mode", "data"), &OpenXRHybridApp::switch_mode, DEFVAL(String()));
+	ClassDB::bind_method(D_METHOD("get_launch_data"), &OpenXRHybridApp::get_launch_data);
+
+	BIND_ENUM_CONSTANT(HYBRID_MODE_NONE);
+	BIND_ENUM_CONSTANT(HYBRID_MODE_IMMERSIVE);
+	BIND_ENUM_CONSTANT(HYBRID_MODE_PANEL);
+}
+
+bool OpenXRHybridApp::is_hybrid_app() const {
+	OS *os = OS::get_singleton();
+	ERR_FAIL_NULL_V(os, false);
+
+	return os->has_feature("godot_openxr_hybrid_app");
+}
+
+OpenXRHybridApp::HybridMode OpenXRHybridApp::get_mode() const {
+	OS *os = OS::get_singleton();
+	ERR_FAIL_NULL_V(os, HYBRID_MODE_NONE);
+
+	if (!is_hybrid_app()) {
+		return HYBRID_MODE_NONE;
+	}
+
+	if (os->has_feature("godot_openxr_panel_app")) {
+		return HYBRID_MODE_PANEL;
+	}
+
+	return HYBRID_MODE_IMMERSIVE;
+}
+
+bool OpenXRHybridApp::switch_mode(HybridMode p_mode, const String &p_data) {
+	ERR_FAIL_COND_V(p_mode == HYBRID_MODE_NONE, false);
+
+	if (!is_hybrid_app()) {
+		return false;
+	}
+
+	Object *godot_openxr_singleton = Engine::get_singleton()->get_singleton("GodotOpenXRHybridAppInternal");
+	if (!godot_openxr_singleton) {
+		return false;
+	}
+
+	return godot_openxr_singleton->call("hybridAppSwitchTo", p_mode, p_data);
+}
+
+String OpenXRHybridApp::get_launch_data() const {
+	if (!is_hybrid_app()) {
+		return "";
+	}
+
+	Object *godot_openxr_singleton = Engine::get_singleton()->get_singleton("GodotOpenXRHybridAppInternal");
+	if (!godot_openxr_singleton) {
+		return String();
```

Comment:
nit: for consistency, we can also return `""` here

Reply:
I went the other way, and made the `""` into `String()` so they are consistently `String()`